src/main/java/Nehsik.java

```diff
@@ -0,0 +1,93 @@
+import java.util.Scanner;
+public class Nehsik {
+    public static final int MAX_TASKS = 100;
+
+    public static void main(String[] args) {
+        displayGreetings();
+        
+        Scanner in = new Scanner(System.in);
+        Task[] taskList = new Task[MAX_TASKS];
+        int taskIndex = 0;
+
+        while (true) {
+            String command = in.nextLine();
+            if (command.equals("list")) {
+                printLine();
+                System.out.println("Here are the tasks in your list:");
+                for (int i = 0; i < taskIndex; i++) {
+                    System.out.println((i + 1) + "." + taskList[i].toString());
+                }
+                printLine();
+            } else if (command.equals("bye")) {
+                break;
+            } else if (command.startsWith("mark")){
+                printLine();
+                System.out.println("Nice! I've marked this task as done:");
+                int taskNum = Integer.parseInt(command.substring(5)) - 1;
```

Comment:
Avoid Magic Numbers! The use of indices like 5 in command.substring(5) and 7 in command.substring(7) directly within the command parsing logic is not ideal. Consider replacing it with named constants.